tracer/src/Datadog.Trace/Configuration/ConfigurationKeys.cs

```diff
@@ -707,6 +707,11 @@ public static class CIVisibility
             /// </summary>
             public const string TotalFlakyRetryCount = "DD_CIVISIBILITY_TOTAL_FLAKY_RETRY_COUNT";
 
+            /// <summary>
+            /// Configuration key for a kill-switch that allows to explicitly disable dynamic instrumentation even if the remote setting is enabled.
+            /// </summary>
+            public const string DynamicInstrumentationEnabled = "DD_CIVISIBILITY_DI_ENABLED";
```

Comment:
Maybe I'm wrong but it seems that what you want here is `ExceptionReplayEnabled` and not DI.